saleor/payment/interface.py

```diff
@@ -0,0 +1,30 @@
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import Dict, Optional
+
+
+@dataclass
+class GatewayResponse:
+    """Dataclass for storing gateway response"""
+
+    is_success: bool
+    kind: str
+    amount: Decimal
+    currency: str
+    transaction_id: str
+    error: Optional[str]
+    raw_response: Optional[Dict[str, str]] = None
+
+
+@dataclass
+class PaymentInformation:
+    """Dataclass for storing all payment information"""
+
+    token: str
+    amount: Decimal
+    currency: str
+    billing: Optional[Dict[str, str]]
```

Comment:
Shouldn't we also have a data class to represent addresses? They're required by most gateways and even now we're using them as dicts, while we exactly know what fields are there.